### PR TITLE
[KOGITO-8167] Adding event state timeout

### PR DESCRIPTION
**Pull request by tiagodolphine**
**JIRA:**

**Description:**

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to setup automatic cherry-pick to another branch ?
</summary>

The cherry-pick action allows to setup automatic cherry-pick from `main` to a specific branch.

To allow it, you will need to add the corresponding label with pattern: `backport-{RELEASE_BRANCH}`.  
For example, if a backport to branch `1.26.x` is needed, then the label should be `backport-1.26.x`.

Once the PR is merged, the action will retrieve the commit and cherry-pick it to the desired branch.

*NOTE: You can still add label after the merge and it should still be cherry-picked.*
</details>

### CODE REVIEW

**Review by wmedvede — COMMENTED** [review]
Hi @tiagodolphine , I think a piece is still missing, see the screenshot, other than this  LGTM. Thanks!

Screenshot from 2022-11-09 10-35-47

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 1]
```diff
@@ -97,6 +97,73 @@ See more details about this state on the link:{spec_doc_url}#switch-date[Serverl
 }
 ----
 
+=== Event state timeout
+
+The `event` state is used to wait for one or more events to be received by the workflow and then continue the execution, if it is not a starting state.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:102]:
```suggestion
The `event` state is used to wait for one or more events to be received by the workflow and then continue the execution if it is not a starting state.
```

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 2]
```diff
@@ -97,6 +97,73 @@ See more details about this state on the link:{spec_doc_url}#switch-date[Serverl
...
+
+The `timeouts` property is used for this state to configure the maximum time the workflow should wait for the defined events to arrive.
+
+If this time is exceeded and the events were not received, the workflow moves to the state defined in the transition property or end state without performing any actions in case of an end state.
+You can see this in the <<event-state, example>>.
+
+See more details about event state timeout on the link:{spec_doc_url}#event-timeout-definition[Serverless Workflow specification].
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:109]:
```suggestion
For more information about event state timeout, see link:{spec_doc_url}#event-timeout-definition[Serverless Workflow specification].
```

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 3]
```diff
@@ -369,6 +436,18 @@ curl -X 'POST' \
 -d '{}'
 ----
 
+* Event
+
+.Creating an Event workflow with timeout
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:441]:
```suggestion
.Creating an event workflow with timeout
```

File: serverlessworkflow/modules/ROOT/pages/getting-started/cncf-serverless-workflow-specification-support.adoc [hunk 4]
```diff
@@ -233,7 +233,7 @@ Alternatively, you can use xref:core/understanding-workflow-error-handling.adoc[
 [[timeouts]]
 == Timeouts
 
-{product_name} has limited support for the Timeouts feature, which covers only *Callback* and *Switch* states with events.
+{product_name} has limited support for the Timeouts feature, which covers only *Callback*, *Switch* and *Event* states.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/getting-started/cncf-serverless-workflow-specification-support.adoc:236]:
```suggestion
{product_name} has limited support for the timeouts feature, which covers only *Callback*, *Switch*, and *Event* states.
```

**Review by kaldesai — APPROVED** [review]
@tiagodolphine I have minor suggestions. Otherwise, your content looks good to me. Please reach out to me if you have any questions about the suggestions. Thank you!

**Review by spolti — APPROVED** [review]
lgtm.

File: serverlessworkflow/modules/ROOT/pages/getting-started/cncf-serverless-workflow-specification-support.adoc [hunk 5]
```diff
@@ -233,7 +233,7 @@ Alternatively, you can use xref:core/understanding-workflow-error-handling.adoc[
 [[timeouts]]
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/getting-started/cncf-serverless-workflow-specification-support.adoc:233]: +1

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc]: +1

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 6]
```diff
@@ -369,6 +436,18 @@ curl -X 'POST' \
 -d '{}'
 ----
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:448]: +1

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 7]
```diff
@@ -97,6 +97,73 @@ See more details about this state on the link:{spec_doc_url}#switch-date[Serverl
 }
 ----
 
+=== Event state timeout
+
+The `event` state is used to wait for one or more events to be received by the workflow and then continue the execution, if it is not a starting state.
+
+The `timeouts` property is used for this state to configure the maximum time the workflow should wait for the defined events to arrive.
+
+If this time is exceeded and the events were not received, the workflow moves to the state defined in the transition property or end state without performing any actions in case of an end state.
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:106]: +1

**Comment by tiagodolphine** [comment]
@kaldesai I added some modifications in the last commit.
@MarianMacik I applied your comments.

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 8]
```diff
@@ -225,6 +298,17 @@ If none of the events arrive before the timeout is overdue, the workflow then mo
 .Switch timeout workflow
 image::core/switch-state-timeouts.svg[]
 
+=== Event workflow
+
+The event example is similar to the switch one but once the execution reaches the state, it waits for one of the configured events, `event1` or `event2`, to arrive, see the <<event-state, event state definition>>.
+
+If none of the configured events arrive before the timeout is overdue, the workflow execution moves to the next state defined in the corresponding `transition` property, skipping the events that were not received in time.
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:305]: +1

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 9]
```diff
@@ -225,6 +298,17 @@ If none of the events arrive before the timeout is overdue, the workflow then mo
...
+
+If one of the events arrives before the timeout is overdue, the workflow then moves to the state defined in `transition`, executing the corresponding event that has arrived in the state definition.
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:307]: +1

**Review by wmedvede — APPROVED** [review]
LGTM, thanks!

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
@kaldesai can you take a second look, please?

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 10]
```diff
@@ -97,6 +97,80 @@ See more details about this state on the link:{spec_doc_url}#switch-date[Serverl
...
 ----
 
+=== Event state timeout
+
+The `event` state is used to wait for one or more events to be received by the workflow and then continue the execution.
+If the event state is a starting state, a new workflow instance is created.
+
+[NOTE]
+====
+Event state is not supported as starting state if `exclusive` flag is set to `false`, therefore timeout is not supported in this case.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:107]:
```suggestion
The event state is not supported as a starting state if the `exclusive` flag is set to `false`, therefore timeout is not supported in this case.
```

**Review by kaldesai — APPROVED** [review]
@tiagodolphine, very minor suggestion. Thank you so much!

**Comment by tiagodolphine** [comment]
@MarianMacik missing your approval, any other comment?
Thanks.

**Review by MarianMacik — APPROVED** [review]

**Pull request merged**